app/policies/default_policy.rb

```diff
@@ -32,16 +33,30 @@ def destroy?
   def admin?
     return true unless Sources::RBAC::Access.enabled?
 
-    user.system.present? || user.user&.org_admin? || write_access?
+    # TODO: remove org_admin after everyone has moved over.
+    # Maybe even remove the `system` check
+    psk_matches? || request.system.present? || request.user&.org_admin? || write_access?
+  end
+
+  def psk_matches?
+    return false if self.class.pre_shared_key.nil?
+
+    self.class.pre_shared_key == key
```

Comment:
In case of receptor, they have array of keys as their env variable (in the vault) and separate key-value in the Vault for each team. What do you think about it?

Reply:
stage-receptor-stage.yml: 
```
- provider: vault-secret
  path: insights/secrets/insights-stage/receptor-stage/receptor-psks
  version: 4
  annotations:
    qontract.recycle: "true"
```

stage-sources-stage.yml:
```
- provider: vault-secret
  path: insights/secrets/insights-stage/receptor-stage/sources-psk
  version: 1
  annotations:
    qontract.recycle: "true"

```

Reply:
Ahhh interesting - I didn't know that, I'll update it to take an array. That way we can follow that implementation.